### PR TITLE
Migrate ipallocator and portallocator to new Events API

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/enhancements/issues/383

```release-note
NONE
```

/kind feature
/priority important-longterm
/sig network

### CODE REVIEW

File: pkg/controlplane/controller.go [hunk 1]
```diff
@@ -172,6 +174,10 @@ func (c *Controller) Start() {
 	repairClusterIPs := servicecontroller.NewRepair(c.ServiceClusterIPInterval, c.ServiceClient, c.EventClient, &c.ServiceClusterIPRange, c.ServiceClusterIPRegistry, &c.SecondaryServiceClusterIPRange, c.SecondaryServiceClusterIPRegistry)
 	repairNodePorts := portallocatorcontroller.NewRepair(c.ServiceNodePortInterval, c.ServiceClient, c.EventClient, c.ServiceNodePortRange, c.ServiceNodePortRegistry)
 
+	c.stopCh = make(chan struct{})
+	repairClusterIPs.Start(c.stopCh)
```
Comment (thockin) [inline pkg/controlplane/controller.go:178]: Why not call these from within RunUntil?  Is it to catch events in RunOnce, I guess?  Maybe a comment here?

Comment (wojtek-t) [reply pkg/controlplane/controller.go:196]: Yes - exactly.

Added a comment here. Together with a TODO to refactor it to simplify the interface and make it easier to run.
Will open an issue for this when this PR merges.

**Comment by wojtek-t** [comment]
@thockin - PTAL

**Pull request merged**